src/xrc/src/api.rs

```diff
@@ -565,18 +590,25 @@ fn handle_fiat_pair(
     env: &impl Environment,
     request: &GetExchangeRateRequest,
 ) -> Result<QueriedExchangeRate, ExchangeRateError> {
-    let timestamp = utils::get_normalized_timestamp(env, request);
+    let requested_timestamp = utils::get_normalized_timestamp(env, request);
     let current_timestamp = env.time_secs();
-    let result = with_forex_rate_store(|store| {
-        store.get(
-            timestamp,
+    let result = if requested_timestamp <= current_timestamp {
```

Comment:
Same here.

Reply:
`requested_timestamp` is normalized to the latest minute. It should not be larger than the `current_timestamp` at this point.